boards/cats_rev1Pro/Core/Src/tasks/task_state_est.c

```diff
@@ -149,46 +143,46 @@ _Noreturn void task_state_est(__attribute__((unused)) void *argument) {
 #endif
 
     /* Check Sensor Readings */
-    err |= check_sensors(&state_data, &elimination);
+    check_sensors(&state_data, &elimination);
 
     /* Write the elimination Data into the global variable */
     global_elimination_data = elimination;
 
     /* Do the preprocessing on the IMU and BARO for calibration */
     /* Only do if we are in MOVING */
-    if (fsm_state.flight_state == MOVING) {
+    if (global_flight_state.flight_state == MOVING) {
       average_data(rolling_imu, &imu_counter, rolling_pressure, &pressure_counter, &elimination, &average_imu,
                    &average_pressure);
     }
 
     /* Do a Kalman Step */
-    kalman_step(&filter, &state_data, &elimination, fsm_state.flight_state);
+    kalman_step(&filter, &state_data, &elimination, global_flight_state.flight_state);
 
     /* write the Data into the global variable */
     global_kf_data.height = (float)filter.x_bar.pData[0];
     global_kf_data.velocity = (float)filter.x_bar.pData[1];
     global_kf_data.acceleration = state_data.acceleration[1];
 
-    /* DO ORIENTATION KALMAN */
-    //    orientation_prediction_step(&orientation_filter, &global_imu[1]);
-    //    orientation_update_step(&orientation_filter, &global_imu[1]);
-    //    log_trace("%ld; %ld; %ld; %ld",
-    //              (int32_t)((float)orientation_filter.x_bar_data[0] * 1000),
-    //              (int32_t)((float)orientation_filter.x_bar_data[1] * 1000),
-    //              (int32_t)((float)orientation_filter.x_bar_data[2] * 1000),
-    //              (int32_t)((float)orientation_filter.x_bar_data[3] * 1000));
-
     uint32_t ts = osKernelGetTickCount();
 
-    /* Log if Sensor is faulty */
-    sensor_info_t sensor_info = {.ts = ts,
-                                 .faulty_baro[0] = elimination.faulty_baro[0],
-                                 .faulty_baro[1] = elimination.faulty_baro[1],
-                                 .faulty_baro[2] = elimination.faulty_baro[2],
-                                 .faulty_imu[0] = elimination.faulty_imu[0],
-                                 .faulty_imu[1] = elimination.faulty_imu[1],
-                                 .faulty_imu[2] = elimination.faulty_imu[2]};
-    record(SENSOR_INFO, &sensor_info);
+    /* DO ORIENTATION KALMAN */
+    orientation_prediction_step(&orientation_filter, &global_imu[1]);
+    orientation_update_step(&orientation_filter, &global_imu[1], &global_magneto);
+    log_trace("%ld; %ld; %ld; %ld", (int32_t)((float)orientation_filter.x_bar_data[0] * 1000),
+              (int32_t)((float)orientation_filter.x_bar_data[1] * 1000),
+              (int32_t)((float)orientation_filter.x_bar_data[2] * 1000),
+              (int32_t)((float)orientation_filter.x_bar_data[3] * 1000));
```

Comment:
You're using float here and float32_t in a previous function. If they are the same we should probably stick to one for consistency